### DISCUSSION TITLE
api access

### DISCUSSION THREAD

**Discussion by feadrick** [Resgrid/Core#43] [Q&A] [answered]
/api/v{version}/Calls/GetActiveCalls
params: version = 3
 i wanted to know on how to access the api above,however i got 401,unauthorized but i already pass the version no which is 3.
so i wanted to what the relation between the api above and the api below,because based on the documentation token from Auth/Validate is used for subsquent request api.
im able to login and get the token from the api

​/api​/v{version}​/Auth​/Validate  -Generates a token that is then used for subsquent requests to the API.

**Comment by ucswift** [comment, marked as answer]
Hey @feadrick!

For the v3 api you'll need to authenticate with the API first `/api/v3/Auth/Validate` passing in `Usr` and `Pass` strings for your login. After that you'll get a payload back with a `Tkn` value that you'll need to pass in as a `Basic` Authorization header value with every request. See an example here for Angular https://github.com/Resgrid/BigBoard/blob/master/src/interceptors/http.interceptor.module.ts#L12.

Other values you'll get on the response payload, `Txd` is when the token expires.
![image](https://user-images.githubusercontent.com/410646/134367335-edd36ba9-b6a8-4d0b-bc5d-5414b6068048.png)
